server/item/writable_book.go

```diff
@@ -0,0 +1,100 @@
+package item
+
+type WritableBook struct {
+	Pages []string
+}
+
+// Page returns a specific page from the book. If the page exists, it will return the content and true, otherwise
+// it will return an empty string and false.
+func (w WritableBook) Page(page uint) (string, bool) {
+	p := int(page)
+	if len(w.Pages) <= p {
+		return "", false
+	}
+	return w.Pages[p], true
+}
+
+// MaxCount always returns 1.
+func (w WritableBook) MaxCount() int {
+	return 1
+}
+
+// Exists checks to see weather a page exists or not
+func (w WritableBook) Exists(page uint) bool {
+	if len(w.Pages) <= int(page) {
+		return false
+	}
+	return true
+}
+
+// Set writes a page to the book, if the page doesn't exist it will be created. It will panic if the
+// text is longer then 256 characters. It will return the newly updated pages.
+func (w WritableBook) Set(page uint, text string) []string {
```

Comment:
SetPage() might be a better name for the method, and same thing as before for the uint parameter. Is it also necessary to return the new set of pages?